src-python/drbench/drbench/plotter.py

```diff
@@ -258,16 +258,21 @@ def __speedup_plot(self, benchmark, device, scaling):
         click.echo(f"writing {fname}")
         db.to_csv(f"{fname}.csv")
 
-        dpl = db.loc[db["Target"] == f"DPL_{device}"]
+        dpl = self.db.copy()
+        dpl = dpl.loc[dpl["Target"] == f"DPL_{device}"]
+        dpl = dpl.loc[dpl["Benchmark"] == benchmark]
         dpl = dpl.loc[dpl["Ranks"] == 1]
         dpl_rtime = dpl["rtime"].values[0]
 
         def points(target):
             p = db.loc[db["Target"] == target]
-            return [p[x_title].values, dpl_rtime / p["rtime"].values, target]
+            total_time = p["rtime"].values / (
```

Comment:
We are scaling the problem size for DR. We pass --weak-scaling to mhp-bench, which makes it scale the problem size by the number of ranks, so we are doing:
```
x: 1, y: time(ref, size:X, ranks:1) / time(DR, size:X * 1, ranks:1)
x: 2, y: time(ref, size:X, ranks:1) * 2 / time(DR, size:X * 2, ranks:2)
x: 3, y: time(ref, size:X, ranks:1) * 3 / time(DR, size:X, * 3 ranks:3)
x: 4, y: time(ref, size:X, ranks:1) * 4 / time(DR, size:X * 4, ranks:4)
```

I don't think it is common to scale the problem size for the reference. You are limited by the memory of a single node so you could not look at large problems or large numbers of nodes.

There are definitely some issues. The wave equation does not scale the problem size with --weak-scaling, so that is a bug that needs to be fixed. Also the CPU speedup numbers do not look right, and I don't know what is the cause.